### PR TITLE
feat(connectors): add pull command

### PR DESCRIPTION
**Pull request by Paveltarno**
> [!NOTE]
> ## Description
>
> Adds a new `base44 connectors pull` command that syncs OAuth connector configurations from Base44 to local files. The command fetches remote integrations and intelligently updates local connector files - adding new ones, updating changed configurations while preserving comments/formatting, and deleting connectors that no longer exist remotely.
>
> ## Related Issue
>
> None
>
> ## Type of Change
>
> - [ ] Bug fix (non-breaking change which fixes an issue)
> - [x] New feature (non-breaking change which adds functionality)
> - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
> - [ ] Documentation update
> - [ ] Refactoring (no functional changes)
> - [ ] Other (please describe):
>
> ## Changes Made
>
> - Added `connectors pull` CLI command with authentication requirement
> - Implemented `writeConnectors()` function in `core/resources/connector/config.ts` for file-based syncing:
>   - Creates/updates connector files from remote data
>   - Deletes local files for removed remote connectors
>   - Uses deep equality checks to preserve comments and formatting when data is unchanged
>   - Resolves connectors by type rather than filename (allows custom filenames)
> - Refactored `readAllConnectors()` to use internal file-entry mapping system for better code reuse
> - Added comprehensive unit tests for `writeConnectors()` covering edge cases (empty lists, custom filenames, unchanged data, directory creation)
> - Added CLI integration tests for the pull command
>
> ## Testing
>
> - [x] I have tested these changes locally
> - [x] I have added/updated tests as needed
> - [x] All tests pass (`npm test`)
>
> ## Checklist
>
> - [x] My code follows the project's style guidelines
> - [x] I have performed a self-review of my own code
> - [x] I have commented my code, particularly in hard-to-understand areas
> - [x] I have made corresponding changes to the documentation (if applicable)
> - [x] My changes generate no new warnings
> - [x] I have updated AGENTS.md if I made architectural changes
>
> ## Additional Notes
>
> The implementation uses `isDeepStrictEqual` to avoid unnecessary file rewrites, preserving user comments and custom formatting in JSONC files. The type-based resolution system allows users to rename connector files (e.g., `my-slack-integration.jsonc` instead of `slack.jsonc`) without breaking functionality - the system matches by `type` field, not filename.
>
> This command complements the existing `connectors push` command, providing bidirectional sync capabilities for connector configurations.
>
> ---
> <sub>🤖 Generated by Claude | 2026-02-15 10:30 UTC</sub>

### CODE REVIEW

File: src/core/resources/connector/api.ts [hunk 1]
```diff
@@ -37,6 +37,10 @@ export async function listConnectors(): Promise<ListConnectorsResponse> {
   return result.data;
 }
 
+export async function fetchConnectors(): Promise<ListConnectorsResponse> {
+  return await listConnectors();
```
Comment (kfirstri) [inline src/core/resources/connector/api.ts]: im confused why not just call listConnectors?

File: src/core/resources/connector/config.ts [hunk 2]
```diff
@@ -34,30 +49,90 @@ export async function readAllConnectors(
...
+              message: `Remove duplicate connectors with type "${entry.data.type}" - only one connector per type is allowed`,
             },
           ],
-        },
+        }
       );
     }
-    types.add(connector.type);
+    typeToEntry.set(entry.data.type, entry);
   }
```
Comment (kfirstri) [inline src/core/resources/connector/config.ts:78]: This  check happens twice, once here and once in readAllconnectors()

**Review by kfirstri — DISMISSED** [review]

File: src/core/resources/connector/config.ts [hunk 3]
```diff
@@ -34,30 +49,82 @@ export async function readAllConnectors(
...
+  return typeToEntry;
+}
+
+export async function readAllConnectors(
+  connectorsDir: string,
+): Promise<ConnectorResource[]> {
+  const entries = await readConnectorFiles(connectorsDir);
+  const typeToEntry = buildTypeToEntryMap(entries);
```
Comment (kfirstri) [inline src/core/resources/connector/config.ts:86]: we don't really need this here, we can just grab the data form `entries` no?
Or is it here for validation there are no two of the same?

**Review by kfirstri — APPROVED** [review]
just one comment :)

File: src/core/resources/connector/config.ts (hunk 3 above)
Comment (Paveltarno) [reply src/core/resources/connector/config.ts:86]: yes exactly, we do validation here

Comment (Paveltarno) [reply src/core/resources/connector/config.ts:86]: `readConnectorFiles` and `buildTypeToEntryMap` can be consolidated into one function but I thought it's nicer to separate the loading from the validation/indexation
